app/src/main/java/com/github/mik629/android/fundamentals/ui/moviedetails/FragmentMovieDetails.kt

```diff
@@ -1,90 +1,97 @@
-import com.github.mik629.android.fundamentals.domain.model.MovieItem
+import com.github.mik629.android.fundamentals.di.movie_details.DaggerMovieDetailsViewModelComponent
 import com.github.mik629.android.fundamentals.ui.global.ActorItemAdapter
+import com.github.mik629.android.fundamentals.ui.utils.buildGlideRequest
+import com.github.mik629.android.fundamentals.ui.utils.setRating
+import com.google.android.material.snackbar.BaseTransientBottomBar
+import com.google.android.material.snackbar.Snackbar
 
-class FragmentMovieDetails : Fragment() {
-    private lateinit var binding: FragmentMovieDetailsBinding
+class FragmentMovieDetails : Fragment(R.layout.fragment_movie_details) {
+    private val binding by viewBinding(FragmentMovieDetailsBinding::bind)
 
     private val glideRequest by lazy {
-        GlideApp.with(this)
-            .asDrawable()
-            .thumbnail(0.1f)
-            .placeholder(R.drawable.ic_image_loading)
-            .fallback(R.drawable.ic_broken_image)
-            .transition(DrawableTransitionOptions.withCrossFade())
-            .fitCenter()
+        buildGlideRequest(this)
     }
-
-    private var movieItem: MovieItem? = null
-
-    override fun onCreate(savedInstanceState: Bundle?) {
-        movieItem = (savedInstanceState ?: arguments)?.getParcelable(ARG_MOVIE_ITEM)
-        super.onCreate(savedInstanceState)
+    private val actorItemAdapter by lazy {
+        ActorItemAdapter(glideRequest)
     }
 
-    override fun onCreateView(
-        inflater: LayoutInflater,
-        container: ViewGroup?,
-        savedInstanceState: Bundle?
-    ): View {
-        binding = FragmentMovieDetailsBinding.inflate(layoutInflater)
-
-        with(binding) {
-            back.setOnClickListener {
-                requireFragmentManager().popBackStack()
-            }
-
-            movieItem?.let {
-                age.text = getString(R.string.movie_min_age, it.minAge)
-                movieTitle.text = it.title
-                genre.text = it.genres.joinToString { genre -> genre.name }
-                ratingLayout.ratingBar.rating = it.rating
-                storyline.text = it.overview
-                reviews.text = getString(R.string.movie_reviews, it.reviews)
-                val actorItemAdapter = ActorItemAdapter(glideRequest)
-                actors.adapter = actorItemAdapter
-                actorItemAdapter.submitList(it.actors)
-            }
-
-            return root
+    private val viewModel: MovieDetailsViewModel by viewModels(
+        factoryProducer = {
+            MovieDetailsViewModelFactory(
+                arguments?.getLong(ARG_MOVIE_ID) ?: 0
+            )
         }
-    }
+    )
 
-    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
-        super.onViewCreated(view, savedInstanceState)
-        movieItem?.let {
-            if (!it.backdrop.isNullOrEmpty()) {
+    override fun onActivityCreated(savedInstanceState: Bundle?) {
```

Comment:
use onViewCreated as it is more known for devs
and onViewCreated name is more familiar comparing with onActivityCreated